cpu/nrf52/radio/nrf802154/nrf802154.c

```diff
@@ -0,0 +1,432 @@
+            DEBUG("[nrf802154] send: unable to do so, packet is too large!\n");
+            mutex_unlock(&txlock);
+            return -EOVERFLOW;
+        }
+        memcpy(&txbuf[len + 1], iolist->iol_base, iolist->iol_len);
+        len += iolist->iol_len;
+    }
+#ifdef MODULE_NETSTATS_L2
+        dev->stats.tx_bytes += len;
+#endif
+
+    /* specify the length of the package. */
+    txbuf[0] = len + IEEE802154_FCS_LEN;
+
+    /* trigger the actual transmission */
+    _enable_tx();
+    DEBUG("[nrf802154] send: putting %i byte into the ether\n", len);
+
+    return len;
+}
+
+static int _recv(netdev_t *dev, void *buf, size_t len, void *info)
+{
+    (void)dev;
+    (void)info;
+
+    size_t pktlen = (size_t)rxbuf[0] - IEEE802154_FCS_LEN;
+
+    /* check if packet data is readable */
+    if (!(_state & NRF_RX_COMPLETE)) {
+        DEBUG("[nrf802154] recv: no packet data available\n");
+        return 0;
+    }
+
+    if (buf == NULL) {
+        if (len > 0) {
+            /* drop packet */
+            DEBUG("[nrf802154] recv: dropping packet of length %i\n", pktlen);
+        }
+        else {
+          /* return packet length */
+          DEBUG("[nrf802154] recv: return packet length: %i\n", pktlen);
+          return pktlen;
+        }
+    }
+    else if (len < pktlen) {
+        DEBUG("[nrf802154] recv: buffer is to small\n");
+        return -ENOBUFS;
+    }
+    else {
+        DEBUG("[nrf802154] recv: reading packet of length %i\n", pktlen);
+        memcpy(buf, &rxbuf[1], pktlen);
+
+#ifdef MODULE_NETSTATS_L2
+        dev->stats.rx_count++;
+        dev->stats.rx_bytes += pktlen;
+#endif
+    }
+
+    _reset_rx();
+
+    return (int)pktlen;
+}
+
+static void _isr(netdev_t *dev)
+{
+    if (!nrf802154_dev.netdev.event_callback) {
+        return;
+    }
+    if (_state & NRF_RX_COMPLETE) {
+        nrf802154_dev.netdev.event_callback(dev, NETDEV_EVENT_RX_COMPLETE);
+    }
+    if (_state & NRF_TX_COMPLETE) {
+        nrf802154_dev.netdev.event_callback(dev, NETDEV_EVENT_TX_COMPLETE);
+        _state &= ~NRF_TX_COMPLETE;
+    }
+}
+
+static int _get(netdev_t *dev, netopt_t opt, void *value, size_t max_len)
+{
+    assert(dev);
+
+    DEBUG("[nrf802154] get: %s\n", netopt2str(opt));
+
+    switch (opt) {
+        case NETOPT_CHANNEL:
+            assert(max_len >= sizeof(uint16_t));
+            *((uint16_t *)value) = nrf802154_dev.chan;
+            return sizeof(uint16_t);
+        case NETOPT_TX_POWER:
+            assert(max_len >= sizeof(int16_t));
+            *((int16_t *)value) = _get_txpower();
+            return sizeof(int16_t);
+
+        default:
+            return netdev_ieee802154_get((netdev_ieee802154_t *)dev,
+                                          opt, value, max_len);
+    }
+}
+
+static int _set(netdev_t *dev, netopt_t opt,
+                const void *value, size_t value_len)
+{
+    assert(dev);
+
+    DEBUG("[nrf802154] set: %s\n", netopt2str(opt));
```

Comment:
Can you apply the following, so we can also compile without `USEMODULE=netopt` and `ENABLE_DEBUG==1`
```suggestion
#ifdef MODULE_NETOPT
    DEBUG("[nrf802154] set: %s\n", netopt2str(opt));
#else
    DEBUG("[nrf802154] set: %d\n", opt);
#endif
```